### PR TITLE
PUD-330 - Use a real database in `RecallRepositoryIntegrationTest`

### PR DESCRIPTION
**Pull request by dazoakley**

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+java adoptopenjdk-16.0.2+7
```
Comment (dazoakley) [inline .tool-versions:1]: This came from Andy's notes on how he set up his macbook: https://mojdt.slack.com/archives/C025S4PG5FT/p1627481715034000?thread_ts=1627383933.012900&cid=C025S4PG5FT

**Review by mr-ken — COMMENTED** [review]
I think there's a better way than that which doesn't require us to run docker-compose in the background.

**Comment by dazoakley** [comment]
Closing this in favour of Ken's (better) solution. :)

**Comment by dazoakley** [comment]
Re-opened as dynamically spinning up a postgres container in CircleCI is problematic (https://github.com/testcontainers/testcontainers-java/issues/1014).

>Effectively the problem is that the containers initiated inside CircleCI are not accessible from the main container.
If any one is interested in this, it would be good to vote for this to be implemented directly from CircleCI.

(ref https://github.com/testcontainers/testcontainers-java/issues/1014#issuecomment-492872176)

**Comment by mr-ken** [comment]
The other solution requires to run the build with a machine executor (rather than docker) - which is a bit slower and more expensive so although it's a less fussy solution (doesn't require docker-compose) this solution is the more pragmatic one.

**Review by mr-ken — APPROVED** [review]

**Pull request merged**